main.js

```diff
@@ -8,154 +8,18 @@
-    };
-  }
-
-  var recastOptions = getRecastOptions(options);
-  var ast = recast.parse(source, recastOptions);
-  var nodePath = new types.NodePath(ast);
-  var programPath = nodePath.get("program");
-
-  if (shouldVarify(source, options)) {
-    // Transpile let/const into var declarations.
-    varifyAst(programPath.node);
-  }
-
-  transform(programPath, options);
-
-  return recast.print(nodePath, recastOptions);
-}
-
-function normalizeOptions(options) {
-  options = utils.defaults(options || {}, {
-    includeRuntime: false,
-    supportBlockBinding: true
+var asyncFunctionSyntax = require("babel-plugin-syntax-async-functions");
+var forOfPlugin = require("babel-plugin-transform-es2015-for-of");
+var babel = require("babel-core");
+
+var regenerator = module.exports = function() {
+  return require("./lib/visit");
+};
+
+regenerator.compile = function(code, opts) {
+  // todo: includeRuntime
+  return babel.transform(code, {
+    plugins: [regenerator, asyncFunctionSyntax, forOfPlugin],
+    sourceType: "script"
   });
-
-  if (!options.esprima) {
-    options.esprima = require("esprima-fb");
-  }
-
-  assert.ok(
-    /harmony/.test(options.esprima.version),
-    "Bad esprima version: " + options.esprima.version
-  );
-
-  return options;
-}
-
-function getRecastOptions(options) {
-  var recastOptions = {
-    range: true
-  };
-
-  function copy(name) {
-    if (name in options) {
-      recastOptions[name] = options[name];
-    }
-  }
-
-  copy("esprima");
-  copy("sourceFileName");
-  copy("sourceMapName");
-  copy("inputSourceMap");
-  copy("sourceRoot");
-
-  return recastOptions;
-}
-
-function shouldVarify(source, options) {
-  var supportBlockBinding = !!options.supportBlockBinding;
-  if (supportBlockBinding) {
-    if (!blockBindingExp.test(source)) {
-      supportBlockBinding = false;
-    }
-  }
-
-  return supportBlockBinding;
-}
-
-function varify(source, options) {
-  var recastOptions = getRecastOptions(normalizeOptions(options));
-  var ast = recast.parse(source, recastOptions);
-  varifyAst(ast.program);
-  return recast.print(ast, recastOptions).code;
-}
-
-function varifyAst(ast) {
-  types.namedTypes.Program.assert(ast);
-
-  var defsResult = require("defs")(ast, {
-    ast: true,
-    disallowUnknownReferences: false,
-    disallowDuplicated: false,
-    disallowVars: false,
-    loopClosures: "iife"
-  });
-
-  if (defsResult.errors) {
-    throw new Error(defsResult.errors.join("\n"))
-  }
-
-  return ast;
-}
-
-// Convenience for just translating let/const to var declarations.
-exports.varify = varify;
-
-// Allow packages that depend on Regenerator to use the same copy of
-// ast-types, in case multiple versions are installed by NPM.
-exports.types = types;
-
-// Transforms a string of source code, returning the { code, map? } result
-// from recast.print.
-exports.compile = compile;
-
-// To modify an AST directly, call require("regenerator").transform(ast).
-exports.transform = transform;
```

Comment:
Removing the `require("regenerator").transform` API is a pretty big breaking change for any non-Babel consumers of this library.